### PR TITLE
feat: setup Satellite basemap selection in MapAndLabel editor

### PR DESCRIPTION
**Pull request by jessicamcinchak**
**Key changes:**
- Adds basemap selection to editor modal
- Adjusts props on public component
- Sets up Mapbox access token env vars throughout
  - [x] `editor.planx.uk/.env` (key 1 is exclusively scoped to localhost:3000 & localhost:5173)
    - **Please note this is no longer committed!!** 
    - I've introduced an `editor.planx.uk/.env.test` and updated `.gitignore` & push/pull `pizza-secret` scripts
    - You may need to run `git rm -r --cached editor.planx.uk/.env` locally if VSCode isn't picking this change up
  - [x] Pulumi staging (key 2 is scoped to our public-facing domains only)
  - [x] Pulumi production (key 2)
  - [x] Github Action secrets (key 2)
Screenshot from 2024-08-28 13-38-22

**Testing:**
- Go to the pizza editor & toggle on the `"TREES"` feature flag
- Create a Map & Label component with "Mapbox Satellite imagery" basemap selected
- Confirm the correct basemap and attribution displays

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -112,41 +112,42 @@ services:
...
+      IDOX_NEXUS_TOKEN_URL: ${IDOX_NEXUS_TOKEN_URL}
+      JWT_SECRET: ${JWT_SECRET}
+      MAPBOX_ACCESS_TOKEN: ${MAPBOX_ACCESS_TOKEN}
+      MINIO_PORT: ${MINIO_PORT}
+      ORDNANCE_SURVEY_API_KEY: ${ORDNANCE_SURVEY_API_KEY}
+      PORT: ${API_PORT}
+      SESSION_SECRET: ${SESSION_SECRET}
+      SLACK_WEBHOOK_URL: ${SLACK_WEBHOOK_URL}
+      UNIFORM_SUBMISSION_URL: ${UNIFORM_SUBMISSION_URL}
+      UNIFORM_TOKEN_URL: ${UNIFORM_TOKEN_URL}
```
Comment (jessicamcinchak) [inline docker-compose.yml:150]: ran "sort lines ascending" on these secrets for readability - only addition is `MAPBOX_ACCESS_TOKEN` :+1:

File: editor.planx.uk/src/@planx/components/List/utils.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export function formatSchemaDisplayValue(
     }
     case "question": {
       const matchingOption = field.data.options.find(
-        (option) => option.data.text === value,
+        (option) => option.data.text === value || option.data.val === value,
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/List/utils.tsx:42]: this is a small bug fix following on from https://github.com/theopensystemslab/planx-new/pull/3521 spotted while testing the List component locally

we need to account for "question" fields that set a `val` OR only `text` - our previous fixes only accounted for one or the other!

File: editor.planx.uk/src/@planx/components/MapAndLabel/Editor.tsx [hunk 3]
```diff
@@ -78,8 +76,68 @@ function MapAndLabelComponent(props: Props) {
...
+                  >
+                    {[
+                      {
+                        id: "OSVectorTile",
+                        title: "Ordnance Survey Vector Tiles",
+                      },
+                      {
+                        id: "MapboxSatellite",
+                        title: "Mapbox Satellite imagery",
+                      },
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/MapAndLabel/Editor.tsx:96]: A nice future improvement here would be to add small static thumbnails of vector tile versus satellite imagery - I don't necessarily expect these names alone to be meaningful to editors! 

I've also intentionally excluded the "OSRaster" & "OSM" options here even though they're valid configs in the map repo - we don't have PlanX use cases for them!

File: editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.tsx [hunk 4]
```diff
@@ -186,11 +188,17 @@ const Root = () => {
             osProxyEndpoint={`${
               import.meta.env.VITE_APP_API_URL
             }/proxy/ordnance-survey`}
-            osCopyright={`© Crown copyright and database rights ${new Date().getFullYear()} OS (0)100024857`}
+            osCopyright={
+              basemap === "OSVectorTile"
+                ? `© Crown copyright and database rights ${new Date().getFullYear()} OS (0)100024857`
+                : ``
+            }
```
[the remark this answers is not in the document]
Comment (jessicamcinchak) [reply editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.tsx:195]: On my radar! These props honestly feel a bit haphazard at this stage - let's see where they stabilise in next ~2 weeks then extract helper functions as needed :+1:

[the remark this answers is not in the document]
Comment (DafyddLlyr) [reply editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.tsx:195]: Makes total sense 👍

**Pull request merged**